### ISSUE TITLE
In BzlMod, overrides are not repo-specific

### ISSUE THREAD

**Issue by Jonpez2** [bazel-contrib/rules_jvm_external#1180]
When using bzlmod, if you invoke 

maven.override(
    name = "repo1",
    coordinates = "foo.bar:baz",
    target = "//foo/bar:baz",
)

and then invoke maven.install twice:

maven.install(
    name = "repo1",
    ...
)

maven.install(
    name = "repo2",
    ...
)

Then the override is applied to both repo1 and repo2.

It is unclear to me whether this is intended behaviour.  I think it isn't, as the parameter 'name' (which I had assumed was intended to point to a 'name' in maven.install(...)) appears to be dropped in the override accumulation logic here https://github.com/bazelbuild/rules_jvm_external/blob/5393477ae7371b21ed0d97fe8202963e206efab8/private/extensions/maven.bzl#L232

My usecase is that I need support a configuration with an overridden dependency, as well as a configuration with the un-overridden dependency.  Pre-bzlmod, I did that with two repos, which did not interfere with each other.

Thank you!

**Comment by Jonpez2** [comment]
Any thoughts on this please?  Thank you!

**Comment by pcj** [comment]
I also got bitten by this.  

TL;DR: it appears intentional, not a bug per-se.

Similar story.  I have the following:

```bazel
maven.install(
    name = "maven_scala",
    artifacts = [
        "com.thesamet.scalapb:compilerplugin_2.12:0.11.17",
    ],
    repositories = ["https://repo1.maven.org/maven2"],
)
```

Here `com.thesamet.scalapb:compilerplugin_2.12:0.11.17` has a dependency on `com.google.protobuf:protobuf-java:3.19.6`, but the following is generated:

```bazel
jvm_import(
    name = "com_thesamet_scalapb_compilerplugin_2_12",
    visibility = ["//visibility:public"],
    tags = [
        "maven_coordinates=com.thesamet.scalapb:compilerplugin_2.12:0.11.17",
        "maven_repository=https://repo1.maven.org/maven2",
        "maven_sha256=a9dc6cc0dbe6ff53a7c914433d5a19711018217b432b385c97778cd4050210d0",
        "maven_url=https://repo1.maven.org/maven2/com/thesamet/scalapb/compilerplugin_2.12/0.11.17/compilerplugin_2.12-0.11.17.jar",
    ],
    jars = [
        "@maven_scala//:v1/https/repo1.maven.org/maven2/com/thesamet/scalapb/compilerplugin_2.12/0.11.17/compilerplugin_2.12-0.11.17.jar",
    ],
    srcjar = "@maven_scala//:v1/https/repo1.maven.org/maven2/com/thesamet/scalapb/compilerplugin_2.12/0.11.17/compilerplugin_2.12-0.11.17-sources.jar",
    deps = [
        "@com_google_protobuf//:protobuf_java",
        "@maven_scala//:com_thesamet_scalapb_protoc_gen_2_12",
        "@maven_scala//:org_scala_lang_modules_scala_collection_compat_2_12",
        "@maven_scala//:org_scala_lang_scala_library",
    ],
)
```

I could not for the life of me figure out how `@com_google_protobuf//:protobuf_java` was showing up here, since I did not ask for any overrides.  To figure it out, I had to debug `rules_jvm_external` itself using a `local_repo_override`.

It turns out I also had in my `MODULE.bazel`:
 
 ```bazel
 bazel_dep(name = "grpc-java", version = "1.64.0", repo_name = "io_grpc_grpc_java")
```

And that module invokes a `maven.override` which forcefully does `maven.override(coordinates="com.google.protobuf:protobuf-java", target="@com_google_protobuf//:protobuf_java")`

can be seen by:

```sh
$ bazel mod show_extension @rules_jvm_external//:extensions.bzl%maven
```

This was quite surprising that the `maven.overrides` declared in a dependency effectively determine it for all `maven.install` rules.

Bzlmod was supposed to eliminate "spooky action at a distance" subtleties from the `WORKSPACE`, but now we have all kinds of new surprises because the `module_extension` implementation functions can and will choose various different version selection behaviors like this one.

**Comment by pcj** [comment]
Perhaps the implementation function in `maven.bzl` should be modified such that overrides (and other configuration) are only shared across `maven_install` instances of they share the same name.  

For example, `grpc-java` uses this:

```bazel
use_repo(
  maven,
  "maven",
)
```

If I wanted to participate in the definition of `@maven//...` declared in grpc-java, I could also choose that name (and hope and pray), otherwise I'd be insulated from it.

**Comment by shs96c** [comment]
We're working on cleaning up how the layering of `install` tags works. Our [current design doc is here](https://docs.google.com/document/d/1zRYmZUzydBo_t3Hdd_DFyzVkKVB4Qxz9xiJa9wk3emI/edit). We can couple this with the new `isolate` attribute of `use_extension` to help prevent modules stomping on each other.

**Comment by ejona86** [comment]
https://github.com/bazel-contrib/rules_jvm_external/pull/1303/files#r1902184110

**Comment by srdo-humio** [comment]
> We're working on cleaning up how the layering of install tags works. Our [current design doc is here](https://docs.google.com/document/d/1zRYmZUzydBo_t3Hdd_DFyzVkKVB4Qxz9xiJa9wk3emI/edit). We can couple this with the new isolate attribute of use_extension to help prevent modules stomping on each other.

The design doc sounds great, but won't isolate end up being a bad idea in some cases?

To give an example, I am depending on `rules_proto_grpc` ,which specifies these dependencies https://github.com/rules-proto-grpc/rules_proto_grpc/blob/5.0.1/modules/java/MODULE.bazel#L25

I'd like to override those to point to my own project's versions of those dependencies, to avoid protobuf version conflicts. If you implement the design doc, I think it should be possible to do that via `maven.override("rules_proto_grpc_java_maven", ...)`. Since `rules_proto_grpc` specifies a custom repo name for `maven.install`, other modules shouldn't be affecting it, even if they have their own `maven.install` commands.

But if `rules_proto_grpc` were to specify `isolate=true`, wouldn't that prevent me from overriding their versions via `maven.override`, so I'd only be able to change their versions via patching their module?
